action/protocol/poll/util.go

```diff
@@ -304,3 +306,47 @@ func shiftProbationList(sm protocol.StateManager) (uint64, error) {
 	}
 	return stateHeight, nil
 }
+
+// setCurrentBlockMeta sets the block meta struct with block meta key
+func setCurrentBlockMeta(
+	sm protocol.StateManager,
+	blockmeta *BlockMeta,
+	height uint64,
+) error {
+	key := blockMetaKey(height)
+	_, err := sm.PutState(blockmeta, protocol.KeyOption(key), protocol.NamespaceOption(protocol.SystemNamespace))
+	return err
+}
+
+// allBlockMetasFromDB returns all latest block meta structs
+func allBlockMetasFromDB(sr protocol.StateReader) ([]*BlockMeta, error) {
+	maxKey := []byte{_blockmeta + 1}
+	stateHeight, iter, err := sr.States(
+		protocol.NamespaceOption(protocol.SystemNamespace),
+		protocol.FilterOption(func(k, v []byte) bool {
+			return bytes.HasPrefix(k, []byte{_blockmeta})
```

Comment:
why is the prefix []byte{_blockmeta} unique?

Reply:
I fixed this issue by having unique prefix by hashing blockMetaPrefix as other creating keys. Prefix == Hash(blockMetaPrefix), so it has no overlap with other keys